sound/soc/sof/topology.c

```diff
@@ -954,10 +937,35 @@ static int sof_parse_tokens(struct snd_soc_component *scomp,
 		/* next array */
 		array = (struct snd_soc_tplg_vendor_array *)((u8 *)array
 			+ asize);
+
+		/* move to next target struct */
+		if (found == count) {
```

Comment:
Didn't you want to change this to `>=`? If I understood you correctly, you said, that currently the ALSA topology compiler cannot generate arrays with elements of different sizes, right? So you should be safe here. But if someone wants to crash the kernel they will make a modified version of the compiler to create invalid topology binaries. So I don't think we can rely on that.

Reply:
ok, changed now to >=